server/src/main/java/org/opensearch/plugins/IdentityPlugin.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.identity.Subject;
+
+/**
+ * Plugin that provides identity and access control for OpenSearch
+ *
+ * @opensearch.experimental
+ */
+public interface IdentityPlugin {
+
+    /**
+     * Get the current subject
+     * */
+    public Subject getSubject();
```

Comment:
Question please, can `getSubject()` return null? If not - could we add Javadocs that subject is required and cannot be null? Thank you.

Reply:
I think just documenting it would be good at this point, since this is a plugin which could be implemented externally, clarifying exceptions would be fine.